tests/algorithms/test_EC.py

```diff
@@ -45,11 +45,68 @@ def test_invalid_algorithm(self):
         with pytest.raises(JWKError):
             ECDSAECKey({'kty': 'bla'}, ALGORITHMS.ES256)
 
+    def test_EC_jwk(self):
+        key = {
+            "kty": "EC",
+            "kid": "bilbo.baggins@hobbiton.example",
+            "use": "sig",
+            "crv": "P-521",
+            "x": "AHKZLLOsCOzz5cY97ewNUajB957y-C-U88c3v13nmGZx6sYl_oJXu9A5RkTKqjqvjyekWF-7ytDyRXYgCF5cj0Kt",
+            "y": "AdymlHvOiLxXkEhayXQnNCvDX4h9htZaCJN34kfmC6pV5OhQHiraVySsUdaQkAgDPrwQrJmbnX9cwlGfP-HqHZR1",
+            "d": "AAhRON2r9cqXX1hg-RoI6R1tX5p2rUAYdmpHZoC1XNM56KtscrX6zbKipQrCW9CGZH3T4ubpnoTKLDYJ_fF3_rJt",
+        }
+
+        ECDSAECKey(key, ALGORITHMS.ES512)
```

Comment:
It would be sensible to check if the keys returned are actually public or private in this test.